[webkitbugspy/webkitscmpy/webkitpy] Replace deprecated datetime methods utcfromtimestamp() and utcnow().

bls1999 opened the pull request:
#### 30b764b0cead09337e77eb03e14afb3fc14798bf
<pre>
[webkitbugspy/webkitscmpy/webkitpy] Replace deprecated datetime methods utcfromtimestamp() and utcnow().
<a href="https://rdar.apple.com/125050750">rdar://125050750</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=271286">https://bugs.webkit.org/show_bug.cgi?id=271286</a>

Reviewed by NOBODY (OOPS!).

The methods datetime.utcfromtimestamp() and datetime.utcnow() are deprecated in python 3.12.
We should remove these calls from the WebKit repo where possible to prevent future issues.

* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/issue.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/bugzilla.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/github.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/radar.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/commit.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/mocks/local/git.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/mocks/local/svn.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/mocks/remote/git_hub.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/mocks/remote/svn.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/pull_request.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/scm_base.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/commit_unittest.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/log_unittest.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/show_unittest.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/svn_unittest.py
* Tools/Scripts/webkitpy/common/checkout/scm/git.py
* Tools/Scripts/webkitpy/performance_tests/perftestsrunner.py
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/30b764b0cead09337e77eb03e14afb3fc14798bf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44682 "check-webkit-style (failure)") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/23769 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/47164 "Built successfully") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/47335 "") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/40687 "Built successfully") 
| [❌ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/46984 "bindings-tests (failure)") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/27807 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/21321 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/47335 "") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/45259 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/20832 "Found unexpected failure with change (failure)") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/39283 "Build is in progress. Recent messages:") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/5/builds/47335 "") | 
| [❌ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/44548 "webkitpy-tests (failure)") | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/39632 "Found unexpected failure with change (failure)") | [❌ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/2729 "") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/41258 "re-run-api-tests (failure)") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/39918 "Found unexpected failure with change (failure)") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/48975 "") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/19640 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/16350 "Found unexpected failure with change (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/48975 "") | 
| [❌ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/44723 "Failed resultsdbpy unit tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/20967 "Built successfully") | | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/48975 "") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/21299 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/20636 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard reviewed (commented):
Some high-level comments here, because in some cases, we want these changes, in others, we don't.

- We shouldn't touch anything in `Source/ThirdParty` unless we _really really_ have to.
- `LayoutTests/imported` will need to be separated, we might need to upstream those to WPT
- `Tools/Scripts/libraries/resultsdbpy` can happen now, although should be a stand-alone PR
- `LayoutTests/http` can also happen now, although should also be a stand-alone PR
- Will need to consult @dewei-zhu and @sundiamonde about `Websites/perf.webkit.org`, that should be a stand-alone PR too
- `webkitbugspy`, `webkitscmpy` and `webkitpy` should stick together, and that's what I'd keep this PR tracking. That being said, all of those still require Python 2 compatibility to some extent, so we might need to keep this as a draft until we drop that requirement

bls1999 committed:
[webkitbugspy/webkitscmpy/webkitpy] Replace deprecated datetime methods utcfromtimestamp() and utcnow().
rdar://125050750
https://bugs.webkit.org/show_bug.cgi?id=271286

Reviewed by NOBODY (OOPS!).

The methods datetime.utcfromtimestamp() and datetime.utcnow() are deprecated in python 3.12.
We should remove these calls from the WebKit repo where possible to prevent future issues.

* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/issue.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/bugzilla.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/github.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/radar.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/commit.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/mocks/local/git.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/mocks/local/svn.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/mocks/remote/git_hub.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/mocks/remote/svn.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/pull_request.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/scm_base.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/commit_unittest.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/log_unittest.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/show_unittest.py
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/svn_unittest.py
* Tools/Scripts/webkitpy/common/checkout/scm/git.py
* Tools/Scripts/webkitpy/performance_tests/perftestsrunner.py
